zerver/lib/outgoing_webhook.py

```diff
@@ -246,8 +246,15 @@ def notify_bot_owner(
     if failure_message:
         notification_message += "\n" + failure_message
     if status_code:
+        if status_code == 407:
+            notification_message += "\nAccess to the outgoing webhook's network is not allowed."
```

Comment:
Maybe a message more like "The configured webhook URL is not network-accessible".  @alexmv may have an even better string, the main edit here is to make clear that the issue is how they configured its URL.

Reply:
Maybe `The URL configured for the webhook is for a private or disallowed network.` ?